pkg/virt-operator/creation/components/crds.go

```diff
@@ -331,6 +331,98 @@ func NewPrometheusRuleSpec(ns string) *promv1.PrometheusRuleSpec {
 							"summary": "More than one virt-api should be running if more than one worker nodes exist.",
 						},
 					},
+					{
+						Record: "num_of_running_virt_controllers",
+						Expr: intstr.FromString(
+							fmt.Sprintf("sum(up{%s='virt-controller', namespace='%s'})", virtv1.AppLabel, ns),
+						),
+					},
+					{
+						Record: "num_of_ready_virt_controllers",
+						Expr: intstr.FromString(
+							fmt.Sprintf("sum(ready_virt_controller{namespace='%s'})", ns),
+						),
+					},
+					{
+						Alert: "LowReadyVirtControllersCount",
+						Expr:  intstr.FromString("num_of_ready_virt_controllers <  num_of_running_virt_controllers"),
+						For:   "5m",
+						Annotations: map[string]string{
+							"summary": "Some virt controllers are running but not ready.",
+						},
+					},
+					{
+						Alert: "NoReadyVirtController",
+						Expr:  intstr.FromString("num_of_ready_virt_controllers == 0"),
+						For:   "5m",
+						Annotations: map[string]string{
+							"summary": "No ready virt-controller was detected for the last 5 min.",
+						},
+					},
+					{
+						Alert: "VirtControllerDown",
+						Expr:  intstr.FromString("num_of_running_virt_controllers == 0"),
+						For:   "5m",
+						Annotations: map[string]string{
+							"summary": "No running virt-controller was detected for the last 5 min.",
+						},
+					},
+					{
+						Alert: "SingleVirtController",
```

Comment:
I guess this is now also redundant when looking at `LowVirtControllersCount` which seems to be an improved version of `SingleVirtController`.